web/src/emoji_picker.ts

```diff
@@ -705,47 +790,50 @@ function handle_status_emoji_clicked(emoji_name) {
     user_status_ui.toggle_clear_message_button();
 }
 
-function handle_composition_emoji_clicked(emoji_name) {
+function handle_composition_emoji_clicked(emoji_name: string): void {
     hide_emoji_popover();
     const emoji_text = ":" + emoji_name + ":";
     // The following check will return false if emoji was not selected in
     // message edit form.
     if (edit_message_id !== null) {
-        const $edit_message_textarea = $(
-            `#edit_form_${CSS.escape(edit_message_id)} .message_edit_content`,
+        const $edit_message_textarea: JQuery<HTMLTextAreaElement> = $(
+            `#edit_form_${CSS.escape(String(edit_message_id))} .message_edit_content`,
         );
         // Assign null to edit_message_id so that the selection of emoji in new
         // message composition form works correctly.
         edit_message_id = null;
-        compose_ui.insert_syntax_and_focus(emoji_text, $edit_message_textarea);
+        compose_ui.insert_syntax_and_focus(emoji_text, $edit_message_textarea, undefined, 0);
     } else {
-        compose_ui.insert_syntax_and_focus(emoji_text);
+        compose_ui.insert_syntax_and_focus(emoji_text, undefined, undefined, 0);
     }
 }
 
-function handle_emoji_clicked($emoji, event) {
+function handle_emoji_clicked(
+    $emoji: JQuery,
+    event: JQuery.ClickEvent<HTMLElement, undefined>,
+): void {
     const emoji_name = $emoji.attr("data-emoji-name");
     const emoji_destination = $emoji
         .closest(".emoji-picker-popover")
         .attr("data-emoji-destination");
 
     switch (emoji_destination) {
         case "reaction": {
-            handle_reaction_emoji_clicked(emoji_name, event);
+            handle_reaction_emoji_clicked(emoji_name ?? "", event);
```

Comment:
Can be asserted IMO. Will `emoji_name` ever be `undefined` in normal running conditions?